src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -372,6 +376,27 @@ public ResponseEntity<byte[]> getLectureAttachment(@PathVariable Long lectureId,
         return buildFileResponse(getActualPathFromPublicPathString(attachment.getLink()), false);
     }
 
+    /**
+     * GET /files/courses/{courseId}/attachments/{attachmentId} : Returns the file associated with the
+     * given attachment ID as a downloadable resource
+     *
+     * @param courseId     The ID of the course that the Attachment belongs to
+     * @param attachmentId the ID of the attachment to retrieve
+     * @return ResponseEntity containing the file as a resource
+     */
+    @GetMapping("files/courses/{courseId}/attachments/{attachmentId}")
+    @EnforceAtLeastEditorInCourse
+    public ResponseEntity<byte[]> getAttachmentFile(@PathVariable Long courseId, @PathVariable Long attachmentId) {
+        Attachment attachment = attachmentRepository.findById(attachmentId)
+                .orElseThrow(() -> new NotFoundAlertException("artemisApp.attachment.pdfPreview.attachmentIDError", ENTITY_NAME, "attachmentIDError"));
+
+        Course course = courseRepository.findByIdElseThrow(courseId);
+
+        checkAttachmentAuthorizationOrThrow(course, attachment);
```

Comment:
This only rechecks the user role in the course, which is already done by `@EnforceAtLeastEditorInCourse`. Instead you should check if the attachment actually is in the course of the specified id. Something like `attachment.lefture.course.equals(course)`

Reply:
Added

Reply:
Done